src/node.h

```diff
@@ -69,6 +69,8 @@ struct cmark_node {
 
   cmark_strbuf string_content;
 
+  cmark_syntax_extension *extension;
```

Comment:
Would it be possible to add this field only to node types that need it? "Custom" or "extension" nodes, for example.